dataactvalidator/scripts/loadFile.py

```diff
@@ -144,6 +144,15 @@ def loadSF133(filename, fiscal_year, fiscal_period, force_load=False):
              "agency_identifier": {"pad_to_length": 3},
              "main_account_code": {"pad_to_length": 4},
              "sub_account_code": {"pad_to_length": 3},
+             # next 3 lines handle the TAS fields that shouldn't
+             # be padded but should still be empty spaces,
+             # at least temporarily, so that downstream formatting
+             # and cleaning functions work correctly. this works
+             # around the fact that sometimes the incoming data
+             # is a single space and sometimes it's blank.
+             "beginning_period_of_availa": {"pad_to_length": 0},
+             "ending_period_of_availabil": {"pad_to_length": 0},
+             "availability_type_code": {"pad_to_length": 0},
```

Comment:
Thanks for the explanation, that does make it much clearer.  I do wonder if we really need to have the pivot index include that many columns, shouldn't 'tas' + 'period' be sufficient to guarantee uniqueness when combined with 'line' as the columns?  Reducing to those would have the side benefit that none of those should be null, which might allow some steps of this process to be simplified.  Or are the extra fields just part of the pivot index to keep them around for the final structure after the melt, and they can't be shifted to values because fill_value can't be limited to a specific value?